modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -0,0 +1,2023 @@
+
+    unsigned int no_fixed_energy_walls = 0;
+    for (unsigned int enum_ind = 0; enum_ind < _energy_wall_types.size(); ++enum_ind)
+      if (_energy_wall_types[enum_ind] == "fixed-temperature" ||
+          _energy_wall_types[enum_ind] == "heatflux")
+        no_fixed_energy_walls += 1;
+
+    if (_wall_boundaries.size() > 0 && _energy_wall_function.size() != no_fixed_energy_walls)
+      paramError("energy_wall_function",
+                 "Size " + std::to_string(_energy_wall_function.size()) +
+                     " is not the same as the number of Dirichlet/Neumann conditions in "
+                     "'energy_wall_types' (" +
+                     std::to_string(no_fixed_energy_walls) + ")");
+
+    if (_inlet_boundaries.size() > 0 && _inlet_boundaries.size() != _energy_inlet_types.size())
+      paramError("energy_inlet_types",
+                 "Size is not the same as the number of inlet boundaries in 'inlet_boundaries'");
+
+    if (_wall_boundaries.size() > 0 && _wall_boundaries.size() != _energy_wall_types.size())
+      paramError("energy_wall_types",
+                 "Size is not the same as the number of wall boundaries in 'wall_boundaries'");
+  }
+}
+
+void
+NSFVAction::checkAmbientConvectionParameterErrors()
+{
+  if (_has_energy_equation)
+  {
+    if (_ambient_convection_blocks.size() && _blocks.size())
+    {
+      for (auto const & block : _ambient_convection_blocks)
+        if (std::find(_blocks.begin(), _blocks.end(), block) == _blocks.end())
+          paramError("ambient_convection_blocks",
+                     "Block '" + block + "' is not present in the block IDs of the module!");
+
+      if (_ambient_convection_blocks.size() != _ambient_convection_alpha.size())
+        paramError("ambient_convection_alpha",
+                   "The number of heat exchange coefficients is not the same as the number of "
+                   "ambient convection blocks!");
+
+      if (_ambient_convection_blocks.size() != _ambient_temperature.size())
+        paramError("ambient_temperature",
+                   "The number of ambient temperatures is not the same as the number of "
+                   "ambient convection blocks!");
+    }
+    else if ((!_ambient_convection_blocks.size() && !_blocks.size()) ||
+             (!_ambient_convection_blocks.size() && _blocks.size()))
+    {
+      if (_ambient_convection_alpha.size() > 1)
+        paramError("ambient_convection_alpha",
+                   "The user should only use one or zero heat exchange coefficient if the ambient "
+                   "convection blocks are not defined!");
+      if (_ambient_convection_alpha.size() != _ambient_temperature.size())
+        paramError("ambient_temperature",
+                   "The number of ambient temperatures is not the same as the number of "
+                   "heat exchange coefficients!");
+    }
+    else if (_ambient_convection_blocks.size() && !_blocks.size())
```

Comment:
This has been changed since _blocks.size() is never 0, we update it to always have at least the default subdomain name. The error checks have been simplified accordingly.